types/webpack/index.d.ts

```diff
@@ -35,6 +35,8 @@ declare function webpack(options: webpack.Configuration[]): webpack.MultiCompile
 
 declare namespace webpack {
     interface Configuration {
+        /** Specify which set of defaults to use. */
+        mode?: "development" | "production";
```

Comment:
Is mode optional?

Reply:
Exceprt from the first paragraph in this article:
https://medium.com/webpack/webpack-4-mode-and-optimization-5423a6bc597a

So webpack 4 added a `mode` option. ***And it’s required.*** *(Actually it’s not required but it will warn you if you omit it.)*

Seems like it will warn if you omit, and default to "production". I took that to mean optional.

Reply:
If it's intended to be required, it would be reasonable to make it non-optional in TS. Might as well point people in the right direction (unless it breaks a lot of existing things).

Reply:
While I don't disagree about pointing people in the right direction, I'm torn over the fact that it will still work if you omit with just a warning.. The warning from webpack itself is already the guide in this case, and as far as the typedef is concerned it is still technically permissable to omit `mode`. Happy to make it required instead though if thats the consensus.

It will break existings things which target v3, but so already does the removal of `CommonChunks` and `module.loaders[]`.